qiita_db/test/test_portal.py

```diff
@@ -1,38 +1,117 @@
 from unittest import TestCase, main
 
 from qiita_core.util import qiita_test_checker
-from qiita_db.portal import (
-    add_study_to_portal, remove_study_from_portal, get_studies_by_portal)
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
+from qiita_db.portal import Portal
 from qiita_db.study import Study
+from qiita_db.analysis import Analysis
+from qiita_db.exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
 
-# Only test if functions available
-if qiita_config.portal == "QIITA":
-    @qiita_test_checker()
-    class TestPortal(TestCase):
-        def setUp(self):
-            self.study = Study(1)
-
-        def test_get_by_portal(self):
-            obs = get_studies_by_portal('EMP')
-            self.assertEqual(obs, set())
-
-            obs = get_studies_by_portal('QIITA')
-            self.assertEqual(obs, {1})
-
-        def test_add_portal(self):
-            add_study_to_portal(self.study, 'EMP')
-            obs = self.study._portals
-            self.assertEqual(obs, ['EMP', 'QIITA'])
-
-        def test_remove_portal(self):
-            with self.assertRaises(ValueError):
-                remove_study_from_portal(self.study, 'QIITA')
-
-            add_study_to_portal(self.study, 'EMP')
-            remove_study_from_portal(self.study, 'EMP')
-            obs = self.study._portals
-            self.assertEqual(obs, ['QIITA'])
+
+@qiita_test_checker()
+class TestPortal(TestCase):
```

Comment:
I missed this before, but can you add tests for the methods that raise warnings?